MainDemo.Wpf/ThemeEditor.xaml.cs

```diff
@@ -0,0 +1,186 @@
+            CustomBaseColorTheme theme = BuildTheme();
+            ApplyTheme(theme);
+            path = null;
+        }
+
+        private void OpenProject_Button_Click(object sender, RoutedEventArgs e)
+        {
+            Microsoft.Win32.OpenFileDialog ofd = new Microsoft.Win32.OpenFileDialog();
+            ofd.Filter = "XAML Files (*.xaml|*.xaml;)";
+            ofd.Multiselect = false;
+            ofd.CheckFileExists = true;
+            ofd.Title = "Select theme project file to edit...";
+            if (ofd.ShowDialog() == false) { return; }
+
+            path = ofd.FileName;
+            try
+            {
+                using (FileStream fs = new FileStream(path, FileMode.Open))
+                {
+                    CustomBaseColorTheme project = (CustomBaseColorTheme)XamlReader.Load(fs);
+                    fs.Close();
+                    fs.Dispose();
+                    MaterialDesignColors.WpfExample.Domain.ThemeEditorViewModel vm = (MaterialDesignColors.WpfExample.Domain.ThemeEditorViewModel)FindResource("VM");
+                    vm.Brushes = MaterialDesignColors.WpfExample.Domain.BrushColor.LoadFromXAML(project);
+                    CustomBaseColorTheme theme = BuildTheme();
+                    ApplyTheme(theme);
+                }
+            }
+            catch //(Exception ex)
+            {
+                commonDialogMessage.Text = "Invalid XAML File";
+                CommonDialog.IsOpen = true;
+            }
+        }
+
+        private void SaveProject_Button_Click(object sender, RoutedEventArgs e)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                SaveProjectAs_Button_Click(sender, e);
+                return;
+            }
+            Flush();
+        }
+
+        private void SaveProjectAs_Button_Click(object sender, RoutedEventArgs e)
+        {
+            Microsoft.Win32.SaveFileDialog sfd = new Microsoft.Win32.SaveFileDialog();
+            sfd.Filter = "XAML Files (*.xaml|*.xaml;)";
+            sfd.Title = "Choose the location to save...";
+            sfd.FileName = "YourOwnThemeColors.xaml";
+            if (sfd.ShowDialog() == false) { return; }
+            path = sfd.FileName;
+            Flush();
+        }
+
+        private void Flush()
+        {
+            using (FileStream fs = new FileStream(path, FileMode.Create, FileAccess.ReadWrite))
+            {
+                var th = BuildTheme();
+                //XamlWriter.Save(th, fs);
+                var xmlWriterSettings = new XmlWriterSettings() { Indent = true, NewLineOnAttributes = true };
+                using (XmlWriter xmlWriter = XmlWriter.Create(fs, xmlWriterSettings))
+                {
+                    th.Clear();
+                    System.Xaml.XamlServices.Save(xmlWriter, th);
+                }
+                fs.Close();
```

Comment:
`.Close()` and `.Dispose()` should not be needed here due to the using statement above.